velox/common/fuzzer/ConstrainedGenerators.cpp

```diff
@@ -286,4 +285,90 @@ getRandomInputGenerator(size_t seed, const TypePtr& type, double nullRatio) {
   return generator;
 }
 
+// CastVarcharInputGenerator
+CastVarcharInputGenerator::CastVarcharInputGenerator(
+    size_t seed,
+    const TypePtr& type,
+    double nullRatio,
+    const TypePtr& castToType)
+    : AbstractInputGenerator(seed, type, nullptr, nullRatio) {
+  castToType_ = castToType;
+}
+
+CastVarcharInputGenerator::~CastVarcharInputGenerator() = default;
+
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+variant CastVarcharInputGenerator::generateRandomGarbageString() {
+  std::string input;
+  std::wstring_convert<std::codecvt_utf8<char16_t>, char16_t> converter;
+
+  // Generate random string.
+  auto randomStr = randString(
+      rng_, rand<size_t>(rng_, 0, 20), {UTF8CharList::ASCII}, input, converter);
+
+  // Further randomize
+  makeRandomStrVariation(
+      input,
+      rng_,
+      RandomStrVariationOptions{
+          CONTROL_CHARACTER_PROBABILITY,
+          ESCAPE_STRING_PROBABILITY,
+          TRUNCATE_PROBABILITY});
+
+  // Return output.
+  return variant(input);
+}
+
+variant CastVarcharInputGenerator::generateValidPrimitiveAsString() {
+  switch (castToType_->kind()) {
+    case TypeKind::BOOLEAN:
+      // For boolean let's alternate between true/false and 1/0.
+      if (coinToss(rng_, .5))
+        return variant(std::to_string(rand<bool>(rng_)));
+      else
+        return variant(rand<bool>(rng_) ? "true" : "false");
+    case TypeKind::INTEGER:
+      return variant(std::to_string(rand<int32_t>(rng_, INT32_MIN, INT32_MAX)));
+    case TypeKind::TINYINT:
+      return variant(std::to_string(rand<int8_t>(rng_, INT8_MIN, INT8_MAX)));
+    case TypeKind::SMALLINT:
+      return variant(std::to_string(rand<int16_t>(rng_, INT16_MIN, INT16_MAX)));
+    case TypeKind::BIGINT:
+      return variant(std::to_string(rand<int64_t>(rng_, INT64_MIN, INT64_MAX)));
+    case TypeKind::HUGEINT:
+      return variant(
+          std::to_string(rand<int128_t>(rng_, INT64_MIN, INT64_MAX)));
+    case TypeKind::REAL:
+      return variant(std::to_string(rand<float>(rng_, FLT_MIN, FLT_MAX)));
+    case TypeKind::DOUBLE:
+      return variant(std::to_string(rand<double>(rng_, FLT_MIN, FLT_MAX)));
+    case TypeKind::TIMESTAMP:
+      return variant(std::to_string(
+          randTimestamp(rng_, FuzzerTimestampPrecision::kMicroSeconds)));
+    case TypeKind::VARCHAR:
+      return generateRandomGarbageString();
```

Comment:
Just call randString() here. At the end of CastVarcharInputGenerator::generate(), we can call makeRandomStrVariation with small probabilities. This would be better because even for the valid input strings, we want sometimes to do a small twist to check whether Velox handle the slightly-wrong input properly.